CefSharp.BrowserSubprocess.Core/JavascriptObjectWrapper.cpp

```diff
@@ -40,13 +62,72 @@ namespace CefSharp
             wrappedproperty->V8Value = javascriptObject;
             wrappedproperty->Bind();
 
-            _wrappedProperties->Add(wrappedproperty);
+            _wrappedProperties->Add(prop->JavascriptName, wrappedproperty);
         }
     }
 
     BrowserProcessResponse^ JavascriptObjectWrapper::GetProperty(String^ memberName)
     {
-        return _browserProcess->GetProperty(_object->Id, memberName);
+        auto response = _browserProcess->GetProperty(_object->Id, memberName);
```

Comment:
This block as a whole is way too complex. Does too many things, needs to be split where possible. Possibly a helper method or two. Desperately needs comments as the single line that was before has now grown into ~50